MediaBrowser.Providers/Plugins/MusicBrainz/MusicBrainzArtistProvider.cs

```diff
@@ -18,257 +10,152 @@
 using MediaBrowser.Controller.Providers;
 using MediaBrowser.Model.Entities;
 using MediaBrowser.Model.Providers;
-using MediaBrowser.Providers.Plugins.MusicBrainz;
-
-namespace MediaBrowser.Providers.Music
-{
-    public class MusicBrainzArtistProvider : IRemoteMetadataProvider<MusicArtist, ArtistInfo>
-    {
-        public string Name => "MusicBrainz";
+using MediaBrowser.Providers.Music;
+using MetaBrainz.MusicBrainz;
+using MetaBrainz.MusicBrainz.Interfaces.Entities;
+using MetaBrainz.MusicBrainz.Interfaces.Searches;
 
-        /// <inheritdoc />
-        public async Task<IEnumerable<RemoteSearchResult>> GetSearchResults(ArtistInfo searchInfo, CancellationToken cancellationToken)
-        {
-            var musicBrainzId = searchInfo.GetMusicBrainzArtistId();
+namespace MediaBrowser.Providers.Plugins.MusicBrainz;
 
-            if (!string.IsNullOrWhiteSpace(musicBrainzId))
-            {
-                var url = "/ws/2/artist/?query=arid:{0}" + musicBrainzId.ToString(CultureInfo.InvariantCulture);
+/// <summary>
+/// MusicBrainz artist provider.
+/// </summary>
+public class MusicBrainzArtistProvider : IRemoteMetadataProvider<MusicArtist, ArtistInfo>, IDisposable
+{
+    private readonly Query _musicBrainzQuery;
 
-                using var response = await MusicBrainzAlbumProvider.Current.GetMusicBrainzResponse(url, cancellationToken).ConfigureAwait(false);
-                await using var stream = await response.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false);
-                return GetResultsFromResponse(stream);
-            }
-            else
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MusicBrainzArtistProvider"/> class.
+    /// </summary>
+    public MusicBrainzArtistProvider()
+    {
+        MusicBrainz.Plugin.Instance!.ConfigurationChanged += (_, _) =>
             {
-                // They seem to throw bad request failures on any term with a slash
-                var nameToSearch = searchInfo.Name.Replace('/', ' ');
-
-                var url = string.Format(CultureInfo.InvariantCulture, "/ws/2/artist/?query=\"{0}\"&dismax=true", UrlEncode(nameToSearch));
-
-                using (var response = await MusicBrainzAlbumProvider.Current.GetMusicBrainzResponse(url, cancellationToken).ConfigureAwait(false))
-                await using (var stream = await response.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false))
-                {
-                    var results = GetResultsFromResponse(stream).ToList();
+                Query.DefaultServer = MusicBrainz.Plugin.Instance.Configuration.Server;
+                Query.DelayBetweenRequests = MusicBrainz.Plugin.Instance.Configuration.RateLimit;
+            };
 
-                    if (results.Count > 0)
-                    {
-                        return results;
-                    }
-                }
+        _musicBrainzQuery = new Query();
```

Comment:
Shouldn't we add a user agent?

Reply:
It's added as default when initializing the plugin:

https://github.com/jellyfin/jellyfin/blob/9062b1c894f0dfb1b8eaee3e5ac05fa1b7dadfac/MediaBrowser.Providers/Plugins/MusicBrainz/Plugin.cs#L31-L34